app/services/CommonTaskListService.scala

```diff
@@ -16,70 +16,82 @@
 package services
 
+import cats.data.EitherT
 import config.AppConfig
-import models.frontend.AllEmploymentData
-import models.tasklist._
+import connectors.errors.ApiError
+import models.frontend.{AllEmploymentData, EmploymentFinancialData}
+import models.mongo.JourneyAnswers
+import models.taskList.TaskStatus.{CheckNow, Completed, InProgress, NotStarted}
+import models.taskList._
 import play.api.Logging
+import repositories.JourneyAnswersRepository
 import uk.gov.hmrc.http.HeaderCarrier
 
 import java.time.Instant
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class CommonTaskListService @Inject()(appConfig: AppConfig,
-                                      service: EmploymentOrchestrationService) extends Logging {
+                                      service: EmploymentOrchestrationService,
+                                      repository: JourneyAnswersRepository) extends Logging {
 
-  def get(taxYear: Int, nino: String, mtditid: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[TaskListSection] = {
+  private def getEmploymentTask(employments: AllEmploymentData,
+                                taxYear: Int,
+                                journeyAnswersOpt: Option[JourneyAnswers]): Option[Seq[TaskListSectionItem]] = {
 
-    service.getAllEmploymentData(nino, taxYear, mtditid).map {
-      case Left(_) => None
-      case Right(employments) => getTask(employments, taxYear)
-    }.map { optionalTask =>
-      TaskListSection(SectionTitle.EmploymentTitle, optionalTask)
-    }
-  }
-
-  private def getTask(employments: AllEmploymentData, taxYear: Int): Option[Seq[TaskListSectionItem]] = {
-
-    val employmentUrl: String = s"${appConfig.employmentFEBaseUrl}/update-and-submit-income-tax-return/employment-income/$taxYear/employment-summary"
+    val baseUrl = s"${appConfig.employmentFEBaseUrl}/update-and-submit-income-tax-return"
+    val employmentUrl: String = s"$baseUrl/employment-income/$taxYear/employment-summary"
 
     def employmentTask(taskStatus: TaskStatus): Option[Seq[TaskListSectionItem]] =
       Some(Seq(TaskListSectionItem(TaskTitle.PayeEmployment, taskStatus, Some(employmentUrl))))
 
-    val hmrcSubmittedOn: Seq[Long] = {
-        employments.hmrcEmploymentData.map { employment =>
-          val submittedOn: String = employment.hmrcEmploymentFinancialData.flatMap(_.employmentData.map(_.submittedOn))
-            .getOrElse(Instant.MIN.toString)
-
-          getSubmittedOnEpoch(submittedOn)
-        }
-      }
-
-    val customerSubmittedOn: Seq[Long] =
-      employments.hmrcEmploymentData.map { employment =>
-        val submittedOn: String  = employment.customerEmploymentFinancialData.flatMap(_.employmentData.map(_.submittedOn))
-          .getOrElse(Instant.MIN.toString)
+    def toSubmittedOn(financialDataOpt: Option[EmploymentFinancialData]): Instant =
+      financialDataOpt.flatMap(
+        _.employmentData.map(
+          data => Instant.parse(data.submittedOn)
```

Comment:
This line might throw runtime error, may be wrap in try , but then we will have to handle the return type as option?Try(Instant.parse(submittedOn)).toOption

Reply:
Presumably, we would want the code to throw a runtime error here if the date can't be parsed as a date? It would only happen if the DES/IF API made a change to the format of this field without informing us in which case I feel like throwing an exception to flag it would be the correct thing to do

Reply:
valid point. i will close this